contributed_definitions/nyaml/NXsample_substrate.yaml

```diff
@@ -0,0 +1,103 @@
+category: base
+doc: |
+  Any information on the sample substrate.
+type: group
+NXsample_substrate(NXobject):
+  material(NX_CHAR):
+    doc: |
+      Material of the substrate in direct contact with the sample.
+  state(NX_CHAR):
+    doc: |
+      Physical state of the substrate, similar options to sample state
+  notes(NXnote):
+    doc: |
+      Further notes.
+  (NXsubstance):
+    doc: |
+     Any substance contained in the sample substrate.  
+  (NX_fabrication):
+    doc: |
+      Name of the sample substrate vendor (company or research group)
+  depends_on(NX_CHAR):
```

Comment:
@rettigl asked for this here is a good example how to NXtransformations in my opinion
the following chain:
1) Use an NXcoordinate_system in an NXcoordinate_system_set instance at the top level
2) if desired add one NXtransformation chain how you have to move your origin into the origin of a coordinate system that is affixed with its origin at the sample
3) from that sample cs origin add another chain of transformation that bring you to some characteristic point where some characteristic point of the sample is hooked into

Reply:
Im still not sure I understand how to use the coordinate system base classes together with NXtransformations. I think I need a practical example.